language/compiler/bytecode_source_map/src/source_map.rs

```diff
@@ -148,9 +171,38 @@ impl<Location: Clone + Eq> FunctionSourceMap<Location> {
     pub fn get_local_name(&self, local_index: u64) -> Option<SourceName<Location>> {
         self.locals.get(local_index as usize).cloned()
     }
+
+    pub fn dummy_function_map(
+        &mut self,
+        module: &CompiledModule,
+        function_def: &FunctionDefinition,
+    ) -> Result<()> {
+        let function_handle = module.function_handle_at(function_def.function);
+        let function_signature = module.function_signature_at(function_handle.signature);
+        let function_code = &function_def.code;
+        let locals = module.locals_signature_at(function_code.locals);
+
+        // Generate names for each type parameter
+        for i in 0..function_signature.type_formals.len() {
+            let name = format!("ty{}", i);
+            self.add_type_parameter((Identifier::new(name)?, Location::default()));
+        }
+
+        // Generate names for each local of the function
+        for i in 0..locals.0.len() {
+            let name = format!("l{}", i);
```

Comment:
I think loc{} may be better, it's really difficult to think about the 'l' otherwise